book/forests.qmd

```diff
@@ -8,29 +8,45 @@ abstract: TODO (150-200 WORDS)
-This value of $20.8$ arises as the sample mean of `mpg` for the $11$ (which can be seen in the box) observations in the training data who were sorted into this terminal node.
-Algorithmically, as splits are always binary, predictions are simply a series of conditional logical statements.
+The decision tree algorithm takes an input, a dataset, selects a variable that is used to partition the data according to some *splitting rule* into distinct non-overlapping datasets or *nodes* or *leaves*, and repeats this step for the resulting partitions, or *branches*, until some criterion has been reached.
+The final nodes are referred to as *terminal nodes*.
+
+This is made clearer by example, (@fig-surv-ranfor) demonstrates a decision tree predicting the miles per gallon (`mpg`) of a car from the `mtcars` dataset [@datamtcars].
+During training, the algorithm identified that the first optimal variable to split the data was the `wt` variable, and that the optimal partition occurs at `wt=2.32`, i.e., one dataset is created for observations with $\code{wt} > 32$ and the other for observations with $\code{wt} \leq 2.32$.
+One of the resulting nodes is left with only $7$ observations and is not split again, the other node is split according to the `disp` variable.
+
+The graphic highlights several core features of decision trees:
+
+1. They can model non-linear effects: The hierarchical structure allows for complex interactions between variables;
+2. They are highly interpretable: it is easy to visualise the tree and see how predictions are made;
+3. They perform variable selection: a very small subset of variables (2 out of 10) were used to train the model by carefully selecting the variables that would lead to the most heterogeneity in the resulting splits -- discussed next.
+
+To understand how random forests work, it is worth looking a bit more into the most important components of decision trees: splitting rules, stopping rules, and terminal node predictions.
 
 ![Demonstrating classification trees using the `mtcars`  [@datamtcars] dataset and the $\pkg{party}$  [@pkgparty] package. Ovals are leaves, which indicate the variable that is being split. Edges are branches, which indicate the cut-off at which the variable is split. Rectangles are terminal nodes and include information about the number of training observations in the node and the terminal node prediction.](Figures/forests/iris_tree.png){#fig-surv-ranfor fig-alt="TODO"}
 
-### Splitting Rules {.unnumbered .unlisted}
+#### Splitting and Stopping Rules {.unnumbered .unlisted}
+
+Precisely how the data partitions/splits are derived and which variables are utilised is determined by the *splitting rule*.
```

Comment:
which variables are utilised is only partially determined by the splitting rule, but also by mtry. 
maybe better to start with assuming only one variabe, explain splitting rule, than go on to generalise to multiple variables

Reply:
hm, I guess trees use all variables as potential split variables?

Reply:
Eh I think it's correct to try all variables when there are few variables, mtry is just to subset to make the algorithm run faster but theoretically you use all

Reply:
yes in trees probably. In forest it serves purposes: a) randomize the tree which improves ensemble properties, b) give variables a chance that have effect, but smaller than some dominating features with large effects (especially if some splitting rule is used that is biased towards features with large variance like Gini importance)

Reply:
Sure but we talk about randomisation later, so I think adding it to trees *could* be confusing?

Reply:
agreed 👍🏻